### PR TITLE
Docs: Plugin E2E - Query editor article

### PR DESCRIPTION
**Pull request by sunker**
**What this PR does / why we need it**:

This is the second article in the `Test a data source plugin` guide. Eventually I'd like to have the following articles:

```
├── test-a-data-source-plugins.md 
│   ├── configuration-editor.md
│   ├── query-editor.md
│   ├── custom-variable-editor.md
│   ├── template-variable-interpolation.md
│   ├── annotation-editor.md
│   ├── alerting.md
```

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: docusaurus/docs/e2e-test-a-plugin/test-a-data-source-plugin/query-editor.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+sidebar_position: 20
+---
+
+## Introduction
+
+The query editor is a central piece of a data source plugin as this is where the query that is used to fetch data is being crafted. Some data source plugins provide rich query editors with various query types targeting different APIs. They can have visual query builders and/or code editors with support for advanced features such as intellisense and autocompletion.
+
+### Testing parts of the query editor in isolation
+
+The query editor in the following example loads regions from a remote API and filters out any GovCloud regions before presenting them in a dropdown. To test that the filtering is working as expected, we can mock the regions resource response and assert that only the non GovCloud regions are shown when the regions dropdown is clicked.
```
Comment (academo) [inline docusaurus/docs/e2e-test-a-plugin/test-a-data-source-plugin/query-editor.md:22]:
```suggestion
The query editor in the following example loads regions via a request to `/regions` and filters out regions containing `-gov-` before presenting them in a dropdown. 

To test that the filtering is working as expected, we can mock the `/regions` resource response and assert that only the regions without `-gov-` in their name are shown when the regions dropdown is clicked.
```

File: docusaurus/docs/e2e-test-a-plugin/test-a-data-source-plugin/query-editor.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  await panelEditPage.datasource.set(ds.name);
+  await panelEditPage.getQueryEditorRow('A').getByText('Query Text').fill('SELECT * FROM dataset');
+  await expect(panelEditPage.refreshPanel()).toBeOK();
+});
+```
+
+### Panel data assertion
+
+In many cases, asserting that the data query response is OK won't be enough to say your data source plugin is functioning properly - you also need to assert on the data that is shown in the panel. Grafana comes with a set of built-in panels, and there's a whole variety of community panels available in the [Plugin catalog](https://grafana.com/grafana/plugins/). All panels can render data differently, so it's impossible to provide a consitent way to assert on the displayed data across all panels. It's therefore recommended to use the `Table` panel for this. The `<page>.panel.data` property returns a [Playwright locator](https://playwright.dev/docs/locators) that resolves element(s) that contain the value(s) that are currently displayed in the Table panel. This means you can use any auto-retrying [matcher](https://playwright.dev/docs/test-assertions#auto-retrying-assertions) that accepts a locator as the receiving type.
```
Comment (sunker) [inline docusaurus/docs/e2e-test-a-plugin/test-a-data-source-plugin/query-editor.md:58]: >...and there's a whole variety of community panels available in the [Plugin catalog](https://grafana.com/grafana/plugins/)

Don't really like this part, so very much open to suggestions here.

File: docusaurus/docs/e2e-test-a-plugin/test-a-data-source-plugin/query-editor.md (hunk 1 above)
Comment (sunker) [reply docusaurus/docs/e2e-test-a-plugin/test-a-data-source-plugin/query-editor.md]: >We can omit specifics such as GovCloud (I don't know what they are but I can tell by the name is about having -gov-)

Yeah good point - it's irrelevant for this article, so let's remove that part about govcloud. 

>My main concern is this example assumes a lot of the knowledge of the person about mocking datasources request. Maybe we can link (if we have) the article to mock datasource requests?

We don't have such an article yet, but I added a sentence that explains what this method does. My changes are based on your suggestions. Let me know what you think.

File: docusaurus/docs/e2e-test-a-plugin/test-a-data-source-plugin/query-editor.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+  - data-source
+  - query editor
+sidebar_position: 20
+---
+
+## Introduction
+
+The query editor is a central piece of a data source plugin as this is where the query that is used to fetch data is being crafted. Some data source plugins provide rich query editors with various query types targeting different APIs. They can have visual query builders and/or code editors with support for advanced features such as intellisense and autocompletion.
+
+### Testing parts of the query editor in isolation
```
[the remark this answers is not in the document]
Comment (josmperez) [reply docusaurus/docs/e2e-test-a-plugin/test-a-data-source-plugin/query-editor.md:20]: Note that doc style guidelines tend to frown upon use of gerunds in favor of siimple present tense (e.g., prefer "test" to "testing").

File: docusaurus/docs/e2e-test-a-plugin/test-a-data-source-plugin/query-editor.md [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+test('data query should return values 1 and 3', async ({ panelEditPage, readProvisionedDataSource }) => {
+  const ds = await readProvisionedDataSource({ fileName: 'datasources.yml' });
+  await panelEditPage.datasource.set(ds.name);
+  await panelEditPage.setVisualization('Table');
+  await expect(panelEditPage.refreshPanel()).toBeOK();
+  await expect(panelEditPage.panel.data).toContainText(['1', '3']);
+});
+```
+
+If you want to assert on what column headers are being displayed in the Table panel, you can use the `<page>.panel.fieldNames` property.
```
Comment (josmperez) [inline docusaurus/docs/e2e-test-a-plugin/test-a-data-source-plugin/query-editor.md:72]:
```suggestion
If you want to assert on what column headers are shown in the `Table` panel, you can use the `<page>.panel.fieldNames` property.
```

**Review by josmperez — APPROVED** [review]
I've made several editorial suggestions, but no showstopper issues. LGTM.

**Review by mckn — APPROVED** [review]
LGTM!

**Pull request merged**